John Haut's Metric


# Total K-12 Enrollment by Neighborhoods

Measuring the total amount of k-12 enrollment each neighborhood has along with how many of the students enrolled in each neighborhood are considered to be gifted. This is one of the metrics that we will be using to deduce what the "best" neighborhood in Pittsburgh is. By calculating the number of gifted students per total enrollment, a new value containing the percentage of gifted students for each neighborhood was produced.This will be used to demonstrate the average Education level within the Neighborhood

# Why K-12 Enrollment

The Total K-12 Enrollment was decided upon to show the capabilities of the educational systems put in place in each neighborhood. If a neighborhood has a lower enrollment it can't be assumed that the number of gifted students in the neighborhood will be lower. Vise Versa neighborhoods with the lowest amount of gifted students cannot be assumed to represent neighborhoods with the lowest enrollment. To narrow in on the best neighborhood, A metric that measures the education quality of each neighborhood needed to be represented. This was done by caculating the gifted student percentage for each of the neighborhoods. 

# The Neighborhoods with the Highest K-12 Enrollment

In [2]:
#EDUCATION AVAILABILITY
#Counting the Enrollment of Schools in each Neighborhood

import pandas as pd 
chip = pd.read_csv("Schools.csv", sep= ",")

#creates a mask of just neighborhoods, gifted students, and total enrollment 
mask = chip[['neighborhoods','gifted', 'total_enrollment_k_12']]

# sorts the values from highest enrollment to lowest 
finished = mask.sort_values(by=['total_enrollment_k_12'], ascending=False)

#prints the highest 10 enrollment neighborhoods 
finished.head(10)

neighborhoods  gifted  \
8                                             Carrick      60   
22                                      Hill District      25   
7                                           Brookline     119   
23                        Homewood North, South, West      27   
15                    East Hills, Wilkinsburg Borough      26   
39        Spring Hill, Summer Hill, Northview Heights      13   
29  Mt. Oliver Borough, Mt. Oliver, Arlington, Arl...      20   
41  Squirrel Hill South, Regent Square, Swisshelm ...     242   
3                                           Beechview      44   
16                              East Liberty, Larimer      37   

    total_enrollment_k_12  
8                    1212  
22                   1136  
7                    1112  
23                   1090  
15                    827  
39                    773  
29                    767  
41                    751  
3                     750  
16                    708

# The Neighborhoods with the Most Gifted K-12 students  

In [3]:
# sorts the values from highest to lowest gifted students 
gifted = mask.sort_values(by = ['gifted'], ascending=False)

#prints 10 neighborhoods with the most gifted students 
gifted.head(10)

neighborhoods  gifted  \
41  Squirrel Hill South, Regent Square, Swisshelm ...     242   
7                                           Brookline     119   
40                                Squirrel Hill North     108   
33                                       Point Breeze      97   
18                                         Greenfield      85   
21                                      Highland Park      83   
8                                             Carrick      60   
3                                           Beechview      44   
4                              Bloomfield, Friendship      43   
6                                    Brighton Heights      39   

    total_enrollment_k_12  
41                    751  
7                    1112  
40                    328  
33                    306  
18                    471  
21                    382  
8                    1212  
3                     750  
4                     291  
6                     566

In [8]:
#make a table here that sorts the data: gifted/ enrolled 

#sets the gifted percentage column = theGifted / theEnrolled * 100
mask.loc['Gifted Percentage'] = mask['gifted'] / mask['total_enrollment_k_12'] *100

#rounds the gifted percentage to two points 
mask.loc['Gifted Percentage'] = mask['Gifted Percentage'].apply(lambda x: float("{:.2f}".format(x)))

#output: sorts the data via gifted percentage 
output = mask.sort_values(by = ['Gifted Percentage'], ascending = False)

KeyError: 'Gifted Percentage'

# Statistics

In [5]:

#print the number of neighborhoods being checked
print("Data set contains a total of " + str(len(mask)) + " neighborhoods located in the Pittsburgh Area \n")

#print the average enrollment of each neighborhood
print("The average enrollment across each neighborhood is: " , str(round(mask['total_enrollment_k_12'].mean(), 2)) , "\n")

#print the average gifted of each neighborhood 
print("The average number of gifted students in a Neighborhood is: ", str(round(mask['gifted'].mean(), 2)), "\n")

#print avg gifted / enrollment of each neighborhood
print("The avg number of gifted students per enrollment is: ", str(round(mask['Gifted Percentage'].mean(),2)),"\n")

Data set contains a total of 45 neighborhoods located in the Pittsburgh Area 

The average enrollment across each neighborhood is:  481.95 

The average number of gifted students in a Neighborhood is:  36.25 



KeyError: 'Gifted Percentage'

# The Neighborhoods with the Highest Gifted/ Enrolled Percentage

In [6]:
#print the top 10 neighborhoods 
output.head(10)

NameError: name 'output' is not defined

# REFLECT ON THE CHOOSEN STATISTIC 

In [7]:
#plot the dataframe on a scatter graph 
output.plot(kind='scatter', x='Gifted Percentage', y='total_enrollment_k_12',
          title="Total Gifted And Enrolled")

NameError: name 'output' is not defined

The following graph represents the data of Gifted percentage by total K-12 enrollment. This means that those further to the right of the graph have the best Percentage of Gifted Students per Total Enrolled. By examining the grouping of the data points it can be seen that for the most part those with the highest students enrolled had the lowest gifted percentages. This can be demonstrated by the Hill District, Homewood North, and Carrick which all have over 1000 total k-12 enrollment, but all three have less than 10% of there students being considered "Gifted". The schools with the best gifted percentage have low to medium total k-12 enrollment. 

# Conclusion


My personal favorite Neighborhood in pittsburgh is the Strip District. This neighborhood has the best food and it has some great activities you can do for fun. Compared to the top neighborhood based on my Metric of education the Strip District is much lower rated than Squirrel Hill North. Since the Strip District only has a Gifted Percentage of 8% it can be concluded that my favorite neighborhood is not as good as Squirrel Hill North with a Percentage of 33%. 